### DISCUSSION TITLE
2024-09-13 온라인 회의

### DISCUSSION THREAD

**Discussion by nyj001012** [jiphyeonjeon-42/backend-nest#40] [General]
참여자 : yena, minsepar, youkim

지난 시간 목표설정 중간점검
이슈 공유

**Comment by nyj001012** [comment]
## 진행도
### histories 스웨거 작성
- 아직 시작 전

## 목표
- ~histories 스웨거 작성 완료~
- ~book-info nest로 변환~

**Comment by scarf005** [comment]
# 진행도

## 명세 작성

- 아직 시작 전

## 목표

- e2e 테스트 설정
- 명세 작성
- 문서화 사이트 열어두기
- compose 파일 만들기

## 명세 예시
- erdcloud 스키마 명세
- DB 어디에서 접근하는지
- compose 어떻게 구성할건지
- https://en.wikipedia.org/wiki/Sequence_diagram

**Comment by parkj12b** [comment]
## 진행도
### users 옮기기
- typeorm-model-generator 로 entity 가져옴
- view + function 긁어오는중

## 목표
- 계속 옮기기 bb ㅠㅠ

**Comment by scarf005** [comment]
# 문서화 해야 하는거

- erdcloud 링크
- DB 덤프
- API 명세
- 시퀀스 다이어그램
  - 어떤 동작 할 수 있는지
  - 예시: 도서 예약 여부, 태그

**Comment by jimin52** [comment]
1. **명세서 작성 여부**
    
    - 오프라인 회의 시 명세 작성 여부 확인.
    - 현재 자료 부재: DB 스키마, 도서관 운영규칙, 백엔드 서버 동작 매핑 관련 자료가 없음.
    - **명세서**는 온보딩 과정에서 중요한 자료로 필요함.
2. **백엔드 문서화**
    
    - **yena**: 백엔드 문서에 명세 반영 제안.
    - **mkdocs**를 활용하여 백엔드 Nest 저장소에 문서화 작업 제안.
3. **문서화 작업 담당자**
    
    - **youkim**: 문서화 작업 담당 희망.
    - 팀 전체 동의로 문서화 작업을 맡기로 결정.
4. **문서화 작업 내용**
    
    - 문서화 작업의 범위 설정:
        - API 명세
        - 도서관 규칙
        - 백엔드 서버의 동작 흐름과 분기 명세
        - 엣지 케이스 정리
    - 기존 서버 API 명세를 따라가면서 진행할지에 대한 논의:
        - API 명세가 난잡한 상황.
        - 현황을 파악 후 취사선택하여 정리하기로 결정.
5. **다음 주 목요일까지 할 일**
    
    - API 현황 파악 및 작업 규모 결정.
6. **참고 자료**
    
    - dumpdb, ERD, 고대 문서(mkdocs, notion 등)를 참고하여 작업 진행.

**Reply by scarf005** [reply]
해보고싶은거

- 도커 컴포즈 설정
	- 시간 나면 devcontainer
- e2e 설정

**Comment by jimin52** [comment]
다음주 목요일까지 목표

예나: tpyeorm Module 관련 학습
지민: 무인화 관련해서 처리할 생각
유킴: e2e, docker compose
민세파: nestjs, typeorm 공부
